system/lib/libc/musl/arch/emscripten/bits/alltypes.h

```diff
@@ -97,28 +97,31 @@ typedef struct { union { int __i[9]; unsigned __s[9]; } __u; } pthread_attr_t;
 #ifdef __EMSCRIPTEN__
 // For mutex implementation in Emscripten, need to use an extra seventh control field
 // to hold a temporary futex wait & wake location, designated as mutex->_m_addr.
-typedef struct { union { int __i[7]; void *__p[7]; } __u; } pthread_mutex_t;
+typedef struct { union { int __i[8]; volatile int __vi[8]; volatile void *__p[8]; } __u; } pthread_mutex_t;
```

Comment:
Since also here musl has added one field, then the Emscripten-specific field needs to also be bumped up by one:

    #ifdef __EMSCRIPTEN__
    #define _m_addr __u.__i[6]
    #endif

should now be `__i[7]` I think. See this commit which originally introduced it: e4d3138215793342b79e6d0d5ed1ba4e6b8a7661